CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+
+  **Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines.
+  The rest of the commit message is then used for this.
+
+  A detailed explanation can be found in this [document](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#).
+
+  **[[back to top]](#TOC)**
+
+### <a name='pullrequests'>Pull requests</a>
+
+
+  - First of all, make sure to base your work on the [appropriate branch][#branches]:
+
+    ```
+    # a bugfix branch for `next` would be prefixed by fix/
+    # a bugfix branch for `master` would be prefixed by hotfix/
+    $ git checkout -b feature/my-feature next
+    ```
+
+  - Write insightful and descriptive commit messages, formatted as [specified][#commits].
+
+  - Please **include the appropriate test cases** for your patch.
+
+  - Make sure all tests pass before submitting your changes. See the
+    [Makefile operations](/README.md#makefile-operations).
+
+  - Make sure the linter does not throw any errors: `make lint`.
+
+  - Rebase your commits. It may be that new commits have been introduced on `next`. Rebasing
+    will update your branch with the most recent code and make your changes easier to review:
+
+    ```
+    $ git fetch
+    $ git rebase origin/next
+    ```
+
+  - Push your changes:
+
+    ```
+    $ git push origin -u feature/my-feature
+    ```
+
+  - Open a pull request against the [appropriate branch][#branches].
+
+  - If we suggest changes:
+  
+    - Please make the required updates (after discussion if any)
+    - Re-run the test suite
+    - Only create new commits if it makes sense. Generally, you will want to 
+      [amend your latest commit or rebase your branch](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
+      after the new changes:
+
+      ```
+      $ git rebase -i next
+      # choose which commits to edit and perform the updates
+      ```
+
+    - Force push to your branch:
+
+      ```
+      $ git push origin feature/my-feature -f
+      ```
+  
+    **[[back to top]](#TOC)**
+
+## <a name='testing'>Testing</a>
+
+  - Use [busted](http://olivinelabs.com/busted) and write lots of tests in a /spec 
+    folder. Separate tests by module.
+  - Use descriptive `describe` and `it` blocks so it's obvious to see what
+    precisely is failing.
+  - Test one functionality per test, repeat for multiple functionalities.
+  - Tests should be atomic, meaning they can only rely on `setup` and `before_each`
+    functionalities, but never on the results or changes by a previous test (even
+    if run on its own it should pass).
+
+  - For the keyword assertions (`nil`, `true`, and `false`) use the `is_xxx` format
+
+    ```lua
+    --bad
+    assert.Nil(something)
+
+    --good
+    assert.is_nil(something)
+    ```
+
+  - Use specific assertions where available. The specific assertion will provide better
+    error messages when they fail, as they better understand the context.
```

Comment:
I actually strongly disagree with this. Granted, they give better error messages, but they act kind of as a "black box", performing assertions on things unrelated to the test at hand (ex: that the body comes from mockbin or httpbin is not relevant to a test, and reduces its atomicity). Most of all, those assertions will hurt us when moving away from mockbin and httpbin, and currently take *quite* some space in our helpers file, and maintenance overhead, all of that for very little value. I never use them for all those reasons, and because `local body = assert.res(200, res)` and `assert.equal("foo", res.headers["bar"])` are more explicit as to what they do, more atomic, and has never prevented me from understanding a failure.